src/document.c

```diff
@@ -473,8 +491,8 @@ FIELD_PREPROCESSOR(geoPreprocessor) {
 
   char *end1 = NULL, *end2 = NULL;
   double lon = strtod(c, &end1);
-  double lat = strtod(pos, &end2);
-  if (*end1 || *end2) {
+  double lat = strtod(pos, &end2); /// failing here
+  if (*end1 || (*end2 && *end2 != '"')) {
```

Comment:
Are we sure end2 + 1 is a valid memory?

Reply:
We assume we use a C string.
We called on it `strtod` and `strpbrk`.

Reply:
Yes so end2 might point to \0 so end2 + 1 is invalid no? Or am I missing something?

Reply:
@MeirShpilraien my suggestion:
` if (*end1 || (*end2 && (*end2 != '"' || *(end2+1)))) {`
if end2 is not null and not quote - it would be an error,
Or if end2 is quote, we can look one char ahead, which could be either the terminating null, or some other char.
If this next char is not the terminating char - it would also be an error.